spec/puppetserver/ca/action/list_spec.rb

```diff
@@ -66,5 +67,40 @@
       expect(out.string).to match(/Signed Certificates:.*foo.*\(SHA256\).*three.*alt names:.*"DNS:foo", "DNS:bar".*/m)
       expect(out.string).to match(/Missing Certificates:.*fake.*/m)
     end
+
+    it 'errors when unknown format option is passed in' do
+      allow(action).to receive(:get_all_certs).and_return(result)
+      exit_code = action.run({'certname' => ['foo', 'baz'], 'format' => 'test'})
+      expect(exit_code).to eq(1)
+    end
+
+    it 'does not throw error when json format option is passed in' do
+      parsed_output = nil
```

Comment:
I don't _think_ this is needed.

Reply:
The `parsed_output` ?

Reply:
setting it ahead of time, yeah. right? since you set it within the `expect`?

Reply:
Apparently that not declaring it there didn't work...I commented that out and the test broke lol.

Reply:
interesting! I think this has to do with the fact that we're using a block in the `expect`. 

I'm actually remembering reading that it's not a best practice to have tests like `expect { blah }.not_to raise_error` - you can just set `parsed_output = JSON.parse(out.string)` and if an error is raised, the test will fail anyway.

Reply:
Neat! Ruby's scoping is pretty interesting and it's something I'm gonna have to catch up on.

Reply:
I think because it's not guaranteed if/when the block will run, you need to declare it first, otherwise it's local to scope of the block. Blocks are [closures](https://www.sitepoint.com/closures-ruby/): they can reference things declared outside of them at the time they were created, and do things like update their values, like we do here. But things created inside of them aren't visible outside, so just declaring `parsed_output` inside the block doesn't let us use it outside of that block.

Reply:
heh like how local variables declared in `if` block over in C/C++ land work